colcon_core/event_handler/__init__.py

```diff
@@ -103,3 +103,49 @@ def apply_event_handler_arguments(extensions, args):
         key = arg[:-1]
         extension = extensions[key]
         extension.enabled = suffix == '+'
+
+
+def format_duration(seconds, *, fixed_decimal_points=None):
+    """
+    Stringify a duration as hours, minutes, seconds and decimal points.
+
+    :param float seconds: The duration in seconds
```

Comment:
Maybe assert seconds > 0? It has some pretty weird results when negative

```
>>> print(format_duration(-1))
-1h 59min 59s
>>> print(format_duration(-3600))
-1h 0min 0s
>>> print(format_duration(-3601))
-2h 59min 59s
```

Reply:
Good point. Added in e2ad19aa7325c408872569676760beb52518db9e.